tests/testthat/test_featsel_filters.R

```diff
@@ -72,3 +72,33 @@ test_that("randomForestSRC_var.select filter handles user choices correctly", {
     more.args = list("randomForestSRC_var.select" = c(method = "vh.imp"))))
   )
 })
+
+test_that("Custom threshold function for filtering works correctly", {
+	biggest_gap <- function(values, diff) {
+	  gap_size = 0
+	  gap_location = 0
+
+	  for (i in (diff+1):length(values)) {
+		gap = values[[i - diff]] - values[[i]]
+		if (gap > gap_size) {
+		  gap_size = gap
+		  gap_location = i - 1
+		}
+	  }
+	  return(gap_location)
+	}
+
+	filt1 = 
+	  makeFilterWrapper(
+		makeLearner(cl = "classif.lda", id = "lda_class", predict.type = "response"), 
+		fw.method = "randomForestSRC_importance",
+		fw.func = biggest_gap,
+		fw.func.args = list("diff" = 1),
+		more.args = list("randomForestSRC_importance" = list(ntree = 50))
+	  )
+	inner = makeResampleDesc("CV", iters = 3, stratify = TRUE)  # Tuning
```

Comment:
Is it necessary to tune here? This will extend the time the test needs to run.

Reply:
No. I have changed this.